### ISSUE TITLE
Use jQuery's event model

### ISSUE THREAD

**Issue by thany** [thomasjbradley/signature-pad#79]
Instead of accepting options like onBeforeValidate and onFormError, the better practice is to take advantage of jQuery's own event model. You would do this by triggering the corresponding event on the element that the plugin was called on, like so:

`this.trigger("beforevalidate", ...);`

Simple enough, right? Then the user (programmer) could attach events to it the way he's used to attaching events:

`$(...).signaturePad().on("beforevalidate", function() { ... });`

Basically.

Is there a specific reason for putting events in the options object?
